flower/utils/results/result.py

```diff
@@ -14,12 +14,12 @@ def __init__(
         *,
         task_id: str,
         status: str,
-        date_done: str,
+        date_done: str | datetime.datetime,
         result: Any,
         traceback: Any,
         # fields with default values may be null when Celery's `result_extended=False`
-        args: Sequence[Any] | None = None,
-        kwargs: dict[str, Any] | None = None,
+        args: Any = None,
+        kwargs: Any = None,
```

Comment:
In theory, `args` should always be a Sequence and `kwargs` should always be a dict (or at least a Mapping). But I ran into some complexities when it comes to deserializing args/kwargs from binary columns from SQLAlchemy records. 

Because results are long-lived, it's possible that a result database table has records that, at the time of creation, used different encoding schemes (e.g. JSON, YAML, pickle, custom). Our result data store makes a best-effort attempt to deserialize these binary values, but ultimately it's very possible that some records could completely fail to deserialize. For example, if someone created a task result row using a "custom" serializer that has since been deleted from the user's Celery codebase, it will be impossible for our result store to deserialize. 

To avoid 500 errors, our result stores can gracefully handle these failures by simply returning the raw `args` or `kwargs` data as it looked when extracted from the data store. Even if we fail to deserialize and you get something ugly like `b'some_kwarg : some_value'`, it's still better than nothing. We don't do much manipulation of `args` or `kwargs` here anyway, so we can afford to sacrifice type specificity for the sake of resilience.

Reply:
I followed some of this but I need to see it live to really internalize it. But what you're saying makes sense and sounds like you've gone through a good evaluation of the edge cases 👍